docs/source/concepts/bento.rst

```diff
@@ -796,76 +796,70 @@ will build a new image on top of the base_image with the following steps:
         bentoml containerize iris_classifier:latest --platform=linux/amd64
 
 
-Docker Template (Danger Zone)
-"""""""""""""""""""""""""""""
+Dockerfile Template (Beta)
+""""""""""""""""""""""""""
 
-The :code:`docker_template` field gives user the full control over how the
-:code:`Dockerfile` was generated in a Bento. Users can use this field to
-customize Bento's Dockerfile instruction set to suits their needs.
+The :code:`dockerfile_template` field gives user the full control over how the
+:code:`Dockerfile` was generated in a Bento. Users can extend the template used for
+generating Dockerfile in a Bento, to suits their needs.
 
-First, create a :code:`Dockerfile.template` file next to your :code:`bentofile.yaml` build file.
-This template file is a mixed between a :code:`Dockerfile` and a :code:`Jinja` template file:
+First, create a :code:`Dockerfile.template` file next to your :code:`bentofile.yaml`
+build file. This file should follow the
+`Jinja2 <https://jinja.palletsprojects.com/en/3.1.x/>`_ template language, and extend
+BentoML's base template and blocks. The template should render a valid
+`Dockerfile https://docs.docker.com/engine/reference/builder/`_. For example:
 
 .. code-block:: dockerfile
 
-   {% extends bento__dockerfile %}
+   {% extends bento_base_template %}
    {% block SETUP_BENTO_COMPONENTS %}
    {{ super() }}
    RUN echo "We are running this during bentoml containerize!"
    {% endblock %}
 
-.. note::
-
-   The template file can have extension :code:`.jinja`, :code:`.j2`, or any
-   extensions that Jinja2 can understand.
-
-Then add the path to the given template file to the :code:`docker_template` field in the :code: `bentofile.yaml`:
+Then add the path to the given template file to the :code:`dockerfile_template` field in
+your :code: `bentofile.yaml`:
 
 .. code:: yaml
 
     docker:
-        docker_template: "./Dockerfile.template"
-
-.. note:: 
+        dockerfile_template: "./Dockerfile.template"
 
-   Although we support any local file path, it is preferred that users put the
-   template file in the project directory.
+Now run :code:`bentoml build` to build a new Bento, it will contain a Dockerfile
+generated with the custom template. To confirm the generated Dockerfile work as
+expected, run :code:`bentoml containerize <bento>` to build a docker image with it.
```

Comment:
```suggestion
Now run :code:`bentoml build` to build a new Bento. It will contain a Dockerfile
generated with the custom template. To confirm the generated Dockerfile works as
expected, run :code:`bentoml containerize <bento>` to build a docker image with it.
```